### DISCUSSION TITLE
commits show up as unsigned when email, passwords, valid access token have been verified. Installed SSH

### DISCUSSION THREAD

**Discussion by Bogdan-coder** [community/community#56890] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

Hey everyone, lately my commits are showing up as unsigned, while the email, passwords, and valid access token have been verified. Installed SSH

No verification status | The commit is not signed.

How can it not be signed? How? If every time I enter the password, the token and the commit goes away

**Comment by airtower-luna** [comment]
"Unverified" means the commit doesn't have a valid signature (this includes none at all). This has nothing to do with the authentication you do when pushing commits, it's about a digital signature recorded in the commit itself, to prove that you made the commit (and not someone else who just put your name and email as author data), and it hasn't been modified.

See [Git Tools - Signing Your Work](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work) for how to create such signatures using GPG (with recent Git versions you can also use SSH keys for signing). You'll also need to tell GitHub what your public keys are, see [Commit signature verification](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification).

**Reply by Bogdan-coder** [reply]
Ok, after I just made myself a GPG key and saved it to GitHub. Now I can't get commits at all - and I see this message - master -> master (rejected non fast forward)

**Reply by airtower-luna** [reply]
That message has nothing to do with commit signing. It seems you're trying to pull, have configured `git pull` to do only fast-forward merges, and the history you're pulling isn't a descendant of the HEAD commit of your local branch. Likely you'll need to either merge or rebase, but that depends on your commit history and what you want to do with it.

**Reply by Bogdan-coder** [reply]
in general after i installed the gpg key reported it to git but still not getting verified

**Reply by Bogdan-coder** [reply]
maybe I shouldn't use SSH and   GPG keys at the same time?

**Reply by Bogdan-coder** [reply]
From GitBash

Gala@DESKTOP-DTLCEI1 MINGW64 ~
$ git config --global --unset gpg.format

Gala@DESKTOP-DTLCEI1 MINGW64 ~
$ gpg --list-secret-keys --keyid-format=long
/c/Users/Gala/.gnupg/pubring.kbx
--------------------------------
sec   rsa2048/F25AFDEF42D2DE6D 2023-05-31 [SC]
      AAC6B81294D77A2E99E5B56BF25AFDEF42D2DE6D
uid                 ---------
ssb   rsa2048/B053637AD406D522 2023-05-31 [E]


Gala@DESKTOP-DTLCEI1 MINGW64 ~
$ git config --global user.signingkey F25AFDEF42D2DE6D

Gala@DESKTOP-DTLCEI1 MINGW64 ~
$ git config --global commit.gpgsign true

**Reply by Bogdan-coder** [reply]
In general, I only verify what I create from the Git website directly, but not with Eclipse

**Reply by airtower-luna** [reply]
Did you upload your public key in your GitHub *settings*, as a signing key? And have you made any new commits, signed with that key?

When you create commits via the web interface, GitHub signs them with their own key. You can see the difference when you check `git log --show-signature`.

**Reply by Bogdan-coder** [reply]
Yes, I uploaded the key and made new commits

**Reply by airtower-luna** [reply]
Can you link any of the commits? It's hard to say what's wrong without it.

**Comment by oliv3r** [comment]
Not sure if OP ever solved his problem, but I see the same issue. I sign my commits with SSH, locally git says everything is a-ok.

```
commit 8b58759f3d1534483b9975ea30dcac72b1e458eb (HEAD -> do_not_leave_worktrees, origin/do_not_leave_worktrees)
Good "git" signature for oliver@schinagl.nl with ED25519 key SHA256:fHlqNgkV403uX3IU8KZenIzRoFhs+D8jQLEbx2GQk6c
Author: Olliver Schinagl <oliver@schinagl.nl>
Date:   Mon Jan 15 16:38:22 2024 +0100
```

The key is perfectly working in the settings:
![image](https://github.com/community/community/assets/1245341/68e7aec6-8882-43aa-adc7-616765b906eb)

Yet, commits are marked as unverified still.
![image](https://github.com/community/community/assets/1245341/c1535c9a-196c-453c-8521-4efefb094cd6)


I wonder though, if this is due to me having several SSH keys, and this is the last in the stack? That would be a bug :)

**Reply by airtower-luna** [reply]
Did you mark the key as a signing key when you uploading it? The "read/write" note looks to me like it might be marked as for authentication.

**Reply by oliv3r** [reply]
I did not, and what's even dumber, is that you cannot set it to 'sign and auth', but have to upload the key twice. From a UX pov this is ridiculous. I get that you may want to use a key only for signing, or only for auth. But I would imagine that the gross of the users would want to use it for both by default.

Thanks for pointing that out! This should resolve it :)
